### PR TITLE
Fix c dial

### PR DESCRIPTION
**Pull request by otaviolcs3**

### CODE REVIEW

File: spiffeid/src/trustdomain.c [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+                                                    err_t *err)
 {
     const char spiffe_scheme[] = "spiffe://";
     const size_t spiffe_scheme_len = sizeof spiffe_scheme - 1;
 
+    string_t str = string_new(uri);
+
     if(!string_contains(str, "://")) {
         // inserts spiffe scheme at beginning of the string
         arrinsn(str, 0, spiffe_scheme_len);
```
Comment (rodrigolc) [inline spiffeid/src/trustdomain.c:13]: does this work? this seems wrong. shouldn't it be 
```suggestion
        arrinsn(str, 0, spiffe_scheme);
```
?

**Review by willallves — APPROVED** [review]

File: spiffeid/src/trustdomain.c [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
         memcpy(str, spiffe_scheme, spiffe_scheme_len);
```
Comment (rodrigolc) [inline spiffeid/src/trustdomain.c:14]: this should be removed if we now allocate a new string and handle the string in place
```suggestion

```

**Review by rodrigolc — CHANGES_REQUESTED** [review]

File: spiffeid/src/trustdomain.c (hunk 2 above)
Comment (rodrigolc) [reply spiffeid/src/trustdomain.c:14]: actually, this one might be correct.

File: spiffeid/src/trustdomain.c (hunk 1 above)
Comment (rodrigolc) [reply spiffeid/src/trustdomain.c:13]: oops, my bad. seems ok then

**Review by rodrigolc — APPROVED** [review]

**Pull request merged**